packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ClientMethodsReturnDescription.java

```diff
@@ -0,0 +1,366 @@
+        }
+        if (responseBodyType == GenericType.FLUX_BYTE_BUFFER && settings.isVanilla()) {
+            // for vanilla + text/xx(e.g. text/powershell) Content-Type
+            syncReturnType = settings.isInputStreamForBinary() ? ClassType.INPUT_STREAM : ClassType.BINARY_DATA;
+        } else {
+            syncReturnType = responseBodyType.getClientType();
+        }
+        syncReturnWithResponse = ResponseTypeFactory.createSyncResponse(operation, syncReturnType, isProtocolMethod,
+            settings, isCustomHeaderIgnored);
+        return new ClientMethodsReturnDescription(operation, asyncRestResponseReturnType, asyncReturnType,
+            syncReturnType, syncReturnWithResponse);
+    }
+
+    /**
+     * Get the {@link ReturnValue} for the given client method type.
+     * <p>
+     * the return value is composed of the client method return type and the JavaDoc for the return type.
+     * </p>
+     *
+     * @param methodType the client method type.
+     * @return the return value.
+     */
+    public ReturnValue getReturnValue(ClientMethodType methodType) {
+        switch (methodType) {
+            case PagingSyncSinglePage:
+                return createReturnValue(syncReturnWithResponse, syncReturnType);
+
+            case PagingAsyncSinglePage:
+                return createReturnValue(asyncRestResponseReturnType, syncReturnType);
+
+            case PagingSync:
+                return createReturnValue(syncReturnType, syncReturnType);
+
+            case PagingAsync:
+                return createReturnValue(asyncReturnType, syncReturnType);
+
+            case SimpleSyncRestResponse:
+                return createReturnValue(syncReturnWithResponse, syncReturnType);
+
+            case SimpleAsyncRestResponse:
+                return createReturnValue(asyncRestResponseReturnType, syncReturnType);
+
+            case SimpleSync:
+                return createReturnValue(syncReturnType, syncReturnType);
+
+            case SimpleAsync:
+                return createReturnValue(asyncReturnType, syncReturnType);
+
+            case LongRunningSync:
+                return createReturnValue(syncReturnType, syncReturnType);
+
+            case LongRunningAsync:
+                return createReturnValue(asyncReturnType, syncReturnType);
+
+            default:
+                throw new IllegalArgumentException("Unsupported method type: " + methodType);
+        }
+    }
+
+    /**
+     * Get the {@link ReturnValue} for the given long-running client method type.
+     * <p>
+     * the return value is composed of the client method return type and the JavaDoc for the return type.
+     * </p>
+     *
+     * @param methodType the long-running client method type.
+     * @param pollingDetails the long-running polling details.
+     * @return the return value.
+     */
+    public ReturnValue getLroBeginReturnValue(ClientMethodType methodType, MethodPollingDetails pollingDetails) {
```

Comment:
Not sure about making a separate method apart from `getReturnValue`. One can accidentally pass in `LongRunningBeginSync` to `getReturnValue` and receive a confusing `Unsupported method type` exception.

Reply:
Right, I initially considered adding the `MethodPollingDetails` parameter to the original `getReturnValue` method, but it required passing null in most cases, making call sites not so nice, that’s why we split it. I'll improve the error message and make `getReturnValue` an overload instead. Since these are internal, I feel providing a better error should suffice.

Reply:
I see. This should suffice, thanks!